examples/src/main/scala/TraceExample.scala

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import cats.Parallel
+import cats.effect.IO
+import cats.effect.IOApp
+import cats.effect.Resource
+import cats.effect.Temporal
+import cats.syntax.all._
+import io.opentelemetry.api.GlobalOpenTelemetry
+import org.typelevel.otel4s.Attribute
+import org.typelevel.otel4s.Otel4s
+import org.typelevel.otel4s.java.OtelJava
+import org.typelevel.otel4s.trace.Tracer
+
+import scala.concurrent.duration._
+
+/*
+An interface for reading a user from a database
+ */
+case class User(name: String, userId: String)
+trait UserDatabase[F[_]] {
+  def readUser(userId: String): F[User]
+}
+
+/*
+An interface for reading a user from a database
+ */
+trait InstitutionServiceClient[F[_]] {
+  def getUserIds(institutionId: String): F[List[String]]
+}
+
+/*
+An interface to compose the getUserIds and readUser methods to get all Users from a single institution.
+ */
+
+trait UsersAlg[F[_]] {
+  def getAllUsersForInstitution(institutionId: String): F[List[User]]
+}
+
+/*
+This database implementation returns a single user out while waiting 30 milliseconds to simulate some time passing.
+The Tracer[F].span is used to span over the simulated database call.
+Note that `span.addAttribute` is used inside of the `.use` to add another attribute to the span with a value that we only know at the time of running the database read.
+ */
+object UserDatabase {
+  def apply[F[_]: Tracer: Temporal]: UserDatabase[F] = new UserDatabase[F] {
+    def readUser(userId: String): F[User] =
+      Tracer[F].span("Read User from DB", Attribute("userId", userId)).use {
+        span =>
+          val returnedUser = User("Zach", "1")
+          Temporal[F].sleep(30.millis) *>
```

Comment:
`flatTap` is more realistic to appear in the actual interaction with the database. 

```suggestion
          Temporal[F].sleep(30.millis)
            .as(User("Zach", "1"))
            .flatTap(user => span.addAttribute(Attribute("User", user.name)))
```

The original option is good if we don't want to overload the example.

Reply:
IMHO extracting it into a method `dbQuery: F[User]` would be even more illustrative.

Reply:
@armanbilge Where would you want that method to live / be scoped to?

Reply:
Not sure, idk if it really matters. It's just about giving it an explicit name, so it's more clear that `sleep+as` is emulating a (slow) query.